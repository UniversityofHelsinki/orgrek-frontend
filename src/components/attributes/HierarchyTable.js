import React from 'react';
import AttributesTable from './AttributesTable';
import Validity from './Validity';
import { useTranslation } from 'react-i18next';
import TableCell from '@mui/material/TableCell';

const HierarchyTable = ({ data, ...props }) => {
  const { t } = useTranslation();

  const columns = [
    {
      label: t('unit'),
      renderCell: (column, item) =>
        // Render unit cell only on the first hierarchy row
        item.rowSpan && (
          <TableCell
            key={column.label}
            rowSpan={item.rowSpan}
            sx={{ verticalAlign: 'top' }}
          >
            {item.unit}
          </TableCell>
        ),
    },
    { label: t('hierarchies'), render: (item) => t(item.hierarchy) },
    {
      label: t('hierarchy_valid'),
      render: (item) => (
        <Validity startDate={item.startDate} endDate={item.endDate} />
      ),
    },
  ];

  const keyFn = (item) =>
    `${item.unit}-${item.hierarchy}-${item.startDate}-${item.endDate}`;

  const hierarchyData = data
    .map((item) =>
      item.hierarchies.map((hierarchy, index) => ({
        unit: item.fullName,
        // Unit name cell spans all hierarchies
        rowSpan: index === 0 && item.hierarchies.length,
        ...hierarchy,
      }))
    )
    .flat();

  return (
    <AttributesTable
      columns={columns}
      keyFn={keyFn}
      data={hierarchyData}
      {...props}
    />
  );
};

export default HierarchyTable;
