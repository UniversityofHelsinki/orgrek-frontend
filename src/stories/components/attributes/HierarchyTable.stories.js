import HierarchyTable from '../../../components/attributes/HierarchyTable';

export default {
  component: HierarchyTable,
};

export const Default = {
  args: {
    summary: 'Yläyksiköt',
    data: [
      {
        id: '1001',
        uniqueId: 10000001,
        startDate: null,
        endDate: null,
        hierarchies: [
          {
            hierarchy: 'virallinen',
            startDate: '2018-12-31',
            endDate: null,
          },
          {
            hierarchy: 'henkilosto',
            startDate: '2018-12-31',
            endDate: null,
          },
          {
            hierarchy: 'opetus',
            startDate: '2018-12-31',
            endDate: null,
          },
          {
            hierarchy: 'tutkimus',
            startDate: '2018-12-31',
            endDate: null,
          },
          {
            hierarchy: 'toiminnanohjaus',
            startDate: '2017-12-31',
            endDate: null,
          },
        ],
        fullName: 'Alayksikkö 1',
        language: 'fi',
      },
      {
        id: '1002',
        uniqueId: 10000002,
        startDate: null,
        endDate: null,
        hierarchies: [
          {
            hierarchy: 'virallinen',
            startDate: null,
            endDate: null,
          },
        ],
        fullName: 'Alayksikkö 2',
        language: 'fi',
      },
    ],
  },
};
