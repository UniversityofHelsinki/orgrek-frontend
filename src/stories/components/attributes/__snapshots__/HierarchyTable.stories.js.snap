// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/attributes/HierarchyTable Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Yksikkö",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Hierarkiat",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Voimassaolo",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Alayksikkö 1",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Talousyksiköt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "31.12.2018 alkaen",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Henkilöstöyksiköt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "31.12.2018 alkaen",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Opetusyksiköt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "31.12.2018 alkaen",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Tutkimusyksiköt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "31.12.2018 alkaen",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Viralliset yksiköt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "31.12.2017 alkaen",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Alayksikkö 2",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Talousyksiköt",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Voimassaoloa ei määritetty",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "stories-components-attributes-hierarchytable--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
